### PR TITLE
Add API calls to SecretStoreClient needed for future PRs

### PR DESCRIPTION
**Pull request by bnevis-i**
Add a number of API calls to the SecretStoreClient
that will be needed to implement Geneva release
security features. These calls support revokation
of previously issued tokens, revokation of the root
token, and generation of new root tokens from
the Vault master key.

Signed-off-by: Bryon Nevis <bryon.nevis@intel.com>

### CODE REVIEW

**Comment by bnevis-i** [comment]
These changes are needed for #1928, #1927, and #1929.

Since no calls have been added to security-secretstore-setup, the testing instructions for this PR are to run the unit tests (`go test`) in `secretstoreclient` and `secretstoreclient\mocks`.

File: internal/security/secretstoreclient/roottoken.go [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+		ResponseObject:       &response,
+	})
+	if response.Complete {
+		*encodedToken = response.EncodedToken
+	}
+	return response.Complete, err
+}
+
+func (vc *vaultClient) rootTokenDecodeToken(encodedToken string, otp string, rootToken *string) error {
+	// otp is base62 ascii pad cast to raw bytes
```
Comment (jim-wang-intel) [inline internal/security/secretstoreclient/roottoken.go]: base64

File: internal/security/secretstoreclient/vault.go [hunk 2]
```diff
@@ -156,3 +155,79 @@ func (vc *vaultClient) CreateToken(token string, parameters map[string]interface
...
+	})
+	*accessors = response.Data.Keys
+	return code, err
+}
+
+func (vc *vaultClient) RevokeAccessor(token string, accessor string) (statusCode int, err error) {
+	parameters := RevokeTokenAccessorRequest{Accessor: accessor}
+	return vc.doRequest(commonRequestArgs{
+		AuthToken:            token,
+		Method:               http.MethodPost,
```
Comment (jim-wang-intel) [inline internal/security/secretstoreclient/vault.go]: not `http.MethodDelete`?

File: internal/security/secretstoreclient/vault.go [hunk 3]
```diff
@@ -156,3 +155,79 @@ func (vc *vaultClient) CreateToken(token string, parameters map[string]interface
...
+		ResponseObject:       &response,
+	})
+	*tokenMetadata = response.Data
+	return code, err
+}
+
+func (vc *vaultClient) RevokeSelf(token string) (statusCode int, err error) {
+	return vc.doRequest(commonRequestArgs{
+		AuthToken:            token,
+		Method:               http.MethodPost,
```
Comment (jim-wang-intel) [inline internal/security/secretstoreclient/vault.go]: not `http.MethodDelete`?

File: internal/security/secretstoreclient/roottoken.go (hunk 1 above)
Comment (bnevis-i) [reply internal/security/secretstoreclient/roottoken.go]: Believe it or not, this is not a typo.

File: internal/security/secretstoreclient/vault.go (hunk 2 above)
Comment (bnevis-i) [reply internal/security/secretstoreclient/vault.go]: Correct.  See https://www.vaultproject.io/api/auth/token/index.html#revoke-a-token-accessor.  Revoke by accessor is defined to be POST.

File: internal/security/secretstoreclient/roottoken.go (hunk 1 above)
Comment (bnevis-i) [reply internal/security/secretstoreclient/roottoken.go]: See https://www.vaultproject.io/api/system/generate-recovery-token.html

File: internal/security/secretstoreclient/vault.go (hunk 3 above)
Comment (bnevis-i) [reply internal/security/secretstoreclient/vault.go]: Correct.  See https://www.vaultproject.io/api/auth/token/index.html#revoke-a-token-self-.  Revoke self is defined to be POST.

**Comment by bnevis-i** [comment]
rebased

File: internal/security/secretstoreclient/roottoken.go (hunk 1 above)
Comment (jim-wang-intel) [reply internal/security/secretstoreclient/roottoken.go]: It is good then

File: internal/security/secretstoreclient/vault.go (hunk 2 above)
Comment (jim-wang-intel) [reply internal/security/secretstoreclient/vault.go]: ok

File: internal/security/secretstoreclient/vault.go (hunk 3 above)
Comment (jim-wang-intel) [reply internal/security/secretstoreclient/vault.go]: ok

**Pull request merged**